### ISSUE TITLE
About supports for timestamp dependency 

### ISSUE THREAD

**Issue by SunBeomSo** [ConsenSys/mythril#1015]
## Description

Detection of timestamp dependency does not seem to work well.

## How to Reproduce
For example, in a very simple example below, Mythril reports nothing.
I think Mythril should report the contract is vulnerable, because there is a dependency from "block.timestamp" to "transfer" instruction (both control- and data dependency).

```
contract Test{
  
    function test(){
        uint n = block.timestamp;
        if (block.timestamp >= now){
            address(0x0).transfer(n);
        }
    }
}
```
Is it a bug or a problem that stems from checking rules? What is the current checking rule of Mythril for detecting timestamp dependency issue? (e.g., there is a dependency between "block.timestamp" to "transfer" instruction, etc)

**Comment by SunBeomSo** [comment]
I used the latest version of Mythril (v. 0.20.4).

**Comment by JoranHonig** [comment]
Thanks for the report!

The variable on the stack seems to be simplified, which causes the module to not know that something is actually a comparison with timestamp.
We should refactor this module so that it annotates variables which have predictable vars, instead of relying on variable names.

**Comment by JoranHonig** [comment]
Should be fixed by #1029

**Comment by b-mueller** [comment]
Yes, #1029 fixed this.

```
$ $MYTH -x lol.sol 
==== Dependence on predictable environment variable ====
SWC ID: 116
Severity: Low
Contract: Test
Function name: test()
PC address: 64
Estimated Gas Usage: 959 - 35240
A control flow decision is made based on a predictable variable.
The block.timestamp environment variable is used in to determine an important control flow decision. Note that the values of variables like coinbase, gaslimit, block number and timestamp are predictable and can be manipulated by a malicious miner. Also keep in mind that attackers know hashes of earlier blocks. Don't use any of those environment variables for random number generation or to make critical control flow decisions.
--------------------
In file: lol.sol:5

if (block.timestamp >= now){
            address(0x0).transfer(n);
        }

--------------------

```
